txpool/txpool.go

```diff
@@ -570,38 +580,61 @@ func (t *TxPool) Underpriced(tx *types.Transaction) bool {
 	return underpriced
 }
 
-func (t *TxPool) Discard(slots uint64, force bool) ([]*types.Transaction, bool) {
+func (t *TxPool) Discard(slotsToRemove uint64, force bool) ([]*types.Transaction, bool) {
 	dropped := make([]*types.Transaction, 0)
-	for t.remoteTxns.Length() > 0 && slots > 0 {
+	for t.remoteTxns.Length() > 0 && slotsToRemove > 0 {
 		tx := t.remoteTxns.Pop()
 		dropped = append(dropped, tx.tx)
 
-		txSlots := numSlots(tx.tx)
-		if slots >= txSlots {
-			slots -= txSlots
+		txSlots := slotsRequired(tx.tx)
```

Comment:
You forgot to eliminate the `else`.

Lines 589-593 can be replaced with
```
txSlots := slotsRequired(tx.tx)
if slotsToRemove < txSlots {
  return dropped, false
}

slotsToRemove -= txSlots
```

Every additional negation we need to do in our heads to figure out what's happening is a wasted thought process :)

Reply:
Refactor that entire piece in https://github.com/0xPolygon/polygon-sdk/pull/213/commits/6a55bb60673518b5e8ff4cbb4a1c7d768e451999